src/hdmf/container.py

```diff
@@ -728,18 +730,22 @@ def _generate_field_html(self, key, value, level, access_code):
             return f'<div style="margin-left: {level * 20}px;" class="container-fields"><span class="field-key"' \
                    f' title="{access_code}">{key}: </span><span class="field-value">{value}</span></div>'
 
-        if hasattr(value, "generate_html_repr"):
+        if isinstance(value, (np.ndarray, h5py.Dataset)):
+            html_content = self._generate_array_html(value, level + 1)
+        elif hasattr(value, "store") and hasattr(value, "shape"):  # Duck typing for zarr array
+            html_content = self._generate_array_html(value, level + 1)
+        elif hasattr(value, "generate_html_repr"):
             html_content = value.generate_html_repr(level + 1, access_code)
-
         elif hasattr(value, '__repr_html__'):
```

Comment:
I also think that maybe that was intended to be `_repr_html_` (single underscore) instead of the double underscore. version we have here?

This is what I found in the ipython documentation:

https://ipython.readthedocs.io/en/stable/config/integrating.html

Reply:
I couldn't find any examples in the code that have the `__repr_html__` with the double underscores so I'm also not sure what the original intention was or where this might be called.

However if we replace this with single underscores `_repr_html_`, other Container objects in the file will display like this with a new header and without the correct margin-left spacing, so I would lean towards leaving for now.
![Screenshot 2024-04-24 at 10 36 53 AM](https://github.com/hdmf-dev/hdmf/assets/40640337/d8d7cc96-171f-420f-a1c5-9d69361feeec)

Reply:
Myserious. OK, we can figure this out in another PR.